### PR TITLE
#292 Test JSON Schema on existing examples from the spec

### PR DESCRIPTION
**Pull request by tfesenko**
JSON Schema for OpenAPI v3 is still under active development. We expect to get a schema for RC2 soon, and I think we will have several more iterations before the final release. Usually, I run JSON Schema validation on my set of spec examples manually, this PR automates it. 
If tests fail it does not mean that the schema is broken, it's a signal to have a closer (human) look at why it fails.
Provide tests validating some examples against the  v3 JSON Schema

@ghillairet , can you please review it?

### CODE REVIEW

File: com.reprezen.swagedit.openapi3.tests/src/com/reprezen/swagedit/openapi3/schema/SchemaValidationTest.xtend [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import org.junit.runners.Parameterized.Parameter
+import org.junit.runners.Parameterized.Parameters
+
+import static org.junit.Assert.*
+import org.junit.Test
+
+@RunWith(typeof(Parameterized))
+class SchemaValidationTest {
+
+	@Parameters(name = "{index}: {1}")
```
Comment (ghillairet) [inline com.reprezen.swagedit.openapi3.tests/src/com/reprezen/swagedit/openapi3/schema/SchemaValidationTest.xtend:31]: Nice

File: com.reprezen.swagedit.openapi3.tests/src/com/reprezen/swagedit/openapi3/schema/SchemaValidationTest.xtend [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	def static Collection<Object[]> data() {
+		val Iterator<URL> specFiles = Iterators::forEnumeration(
```
Comment (ghillairet) [inline com.reprezen.swagedit.openapi3.tests/src/com/reprezen/swagedit/openapi3/schema/SchemaValidationTest.xtend:33]: Not an issue but It needs to be run as junit plugin test, when I try to run it as normal junit test I get

```
java.lang.NullPointerException
	at com.reprezen.swagedit.openapi3.schema.SchemaValidationTest.data(SchemaValidationTest.java:39)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.runners.Parameterized.allParameters(Parameterized.java:292)
	at org.junit.runners.Parameterized.<init>(Parameterized.java:282)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
	at org.junit.internal.builders.AnnotatedBuilder.buildRunner(AnnotatedBuilder.java:29)
	at org.junit.internal.builders.AnnotatedBuilder.runnerForClass(AnnotatedBuilder.java:21)
	at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:59)
	at org.junit.internal.builders.AllDefaultPossibilitiesBuilder.runnerForClass(AllDefaultPossibilitiesBuilder.java:26)
	at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:59)
	at org.junit.internal.requests.ClassRequest.getRunner(ClassRequest.java:26)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestLoader.createUnfilteredTest(JUnit4TestLoader.java:84)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestLoader.createTest(JUnit4TestLoader.java:70)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestLoader.loadTests(JUnit4TestLoader.java:43)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:444)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:678)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)
```

**Review by ghillairet — APPROVED** [review]
Looks fine, I left a comment about having to run a test as plugin test, maybe you always run them like that. I know all other tests can be run as normal junit tests.

Also in ValidatorTest, the test `invlaidAlias` fails, is it expected?

**Comment by tfesenko** [comment]
@ghillairet , I will address your comments.
Re: test failures. I just had the same problem on the master branch. The problem is that `mvn clean` does not empty the `xtend-gen` folder and stale classes may cause unexpected results. I will update our pom.xml to clean up xtend resources.

**Comment by tfesenko** [comment]
@ghillairet , I addressed some of your comments, but I don't have the time to make them run as regular JUnit tests now. I created #343 to address it later.

**Pull request merged**